roles/postgresql/defaults/main.yml

```
@@ -119,17 +119,22 @@ postgresql_generated_databases: >-
 postgresql_additional_databases: []
   # - name: bar
   #   owner: foo
+  #   schema_generated: yes # default (auto generate schema list)   
   #   hba_entry_generated: yes # default (activate auto generated pg_hba entry)
   #   hba_type: host # default
   #   hba_addresses:
   #     - all # default
 
 postgresql_databases: "{{ postgresql_generated_databases + postgresql_additional_databases }}"
 
+#
+# Hive MetaStore use of "public" schema is hardcoded in its scripts, therefore we do not create a specific schema for Hive (issue #28)
+# "schema_generated" is set to "no" in "hive" database
+#
 postgresql_generated_schemas: >-
   [
     {%- for db in postgresql_databases -%}
-      {%- if db.owner is defined -%}
+      {%- if db.owner is defined and (not db.schema_generated is defined or not db.schema_generated|length or db.schema_generated == "yes") -%}
```

Can you test this condition if it works ?
```suggestion
      {%- if db.owner is defined and (db.schema_generated is not defined or db.schema_generated) -%}
```
You can see this pattern here https://github.com/TOSIT-IO/tdp-collection-prerequisites/blob/b2de13e2b70b71e9a98ae8cae1b57c519728d754/roles/postgresql/defaults/main.yml#L157

---

For ```hive```, if I use the following condition:
```
      {%- if db.owner is defined and (db.schema_generated is not defined or db.schema_generated) -%}
```
- if ```schema_generated``` is not set in database, ```hive``` schema is generated
- if ```"schema_generated": ""``` set to empty in database, ```hive``` schema is not generated
- if ```"schema_generated": "yes"``` is set to "yes" in database, ```hive``` schema is generated
- if ```"schema_generated": "no"``` is set to "no" in database, ```hive``` schema is generated

---

`yes` and `no` are not a boolean for a JSON, the Jinja code in the variable output a JSON so JSON type should be use and a boolean in JSON is `true` or `false`.

>     * if `schema_generated` is not set in database, `hive` schema is generated
I agree
>     * if `"schema_generated": ''` set to empty in database, `hive` schema is not generated
`schema_generated` is a boolean, `''` is empty string which is not a boolean, you can ignore this case.
>     * if `"schema_generated": true` is set to "True" in database (to replace "yes"), `hive` schema is generated
I agree
>     * if `"schema_generated": false` is set to "False" in database (to replace "no"), `hive` schema is not generated
I agree

So the condition I propose works no ?

---

Yes, your condition works for the flags ```schema_generated``` and ```hba_entry_generated``` set to boolean.
So I think it is worth to clarify in lines 122 and 123 that the flags are boolean (set to 'true' and not to 'yes'):
```
  #   schema_generated: true # default (auto generate schema list)   
  #   hba_entry_generated: true # default (activate auto generated pg_hba entry)
```

---

Yes, update the doc to use `true` or `false`.